plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -1,17 +1,43 @@
 package io.jenkins.plugins.remotingkafka;
 
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.SchemeRequirement;
+import com.cloudbees.plugins.credentials.matchers.IdMatcher;
 import hudson.Extension;
+import hudson.model.Item;
+import hudson.security.ACL;
 import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
 import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
 import net.sf.json.JSONObject;
 import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.AncestorInPath;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Collections;
+import java.util.List;
+
 @Extension
 public class GlobalKafkaConfiguration extends GlobalConfiguration {
-    private String connectionURL;
+    public static final SchemeRequirement KAFKA_SCHEME = new SchemeRequirement("kafka");
+    private String credentialsId;
+    private String brokerURL;
     private String zookeeperURL;
+    private String username;
+    private String password;
+    private String sslTruststoreLocation;
+    private String sslTruststorePassword;
+    private String sslKeystoreLocation;
+    private String sslKeystorePassword;
+    private String sslKeyPassword;
```

Comment:
🐛 You should use `hudson.util.Secret` at least, otherwise credentials get stored in plain text. But generally I would recommend switching it to credentials as well so that Jenkins admins can get Kafka authorized without using external key store (if possible, of course)

Reply:
Internal key store may be a separate step. But generally Jenkins supports SSH credentials: https://wiki.jenkins.io/display/JENKINS/SSH+Credentials+Plugin